document_page_reference/models/document_page.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2019 Creu Blanca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+import re
+
+from jinja2.sandbox import SandboxedEnvironment
+from markupsafe import Markup
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools import html_escape
+
+_logger = logging.getLogger(__name__)
+env = SandboxedEnvironment(autoescape=False)
+
+
+class DocumentPage(models.Model):
+    _inherit = "document.page"
+    _description = "Document Page"
+
+    reference = fields.Char(
+        help="Used to find the document, it can contain letters, numbers and _"
+    )
+    content_parsed = fields.Html(
+        "Parsed Content", compute="_compute_content_parsed", sanitize=False, store=True
+    )
+
+    def get_formview_action(self, access_uid=None):
+        res = super().get_formview_action(access_uid)
+        view_id = self.env.ref("document_page.view_wiki_form").id
+        res["views"] = [(view_id, "form")]
+        return res
+
+    @api.depends("content")
+    def _compute_content_parsed(self):
+        for record in self:
+            try:
+                raw = record.content
+                references = re.findall(r"\$\{([^}]+)\}", raw)
+                for ref in references:
+                    if not re.match(r"^[a-zA-Z_][a-zA-Z0-9_]*$", ref):
+                        raise ValidationError(_("Invalid reference: %s") % ref)
+                converted = re.sub(r"\$\{([\w_]+)\}", r"{{ resolve('\1') }}", raw)
+                template = env.from_string(converted)
+                rendered = template.render(resolve=record._resolve_reference)
+                record.content_parsed = rendered
+            except Exception as e:
+                _logger.warning("Render failed for %s: %s", record.id, e)
```

Comment:
```suggestion
                _logger.info("Render failed for %s: %s", record.id, e)
```
This change is necessary for the CI to be green.